drivers/filesystems/fastfat/create.c

```diff
@@ -791,6 +789,10 @@ VfatCreateFile(
                 Attributes |= FILE_ATTRIBUTE_ARCHIVE;
             }
             vfatSplitPathName(&PathNameU, NULL, &FileNameU);
+
+            if (IsDotOrDotDot(FileNameU))
+                return STATUS_OBJECT_NAME_INVALID;
```

Comment:
You'll need to clean up here and release the FCB (and, I suppose, add to statistics). See the failure path a few lines above.

Reply:
Fixed. Thanks.